### PR TITLE
Add a basic WebGPU example

### PR DESCRIPTION
**Pull request by aleino-nv**
- Add a simple example of using WebGPU and Slang together

https://github.com/shader-slang/slang/issues/5656

### CODE REVIEW

File: examples/wgpu-html5/example.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
+"use strict";
+
+let Example = {
```
Comment (csyonghe) [inline examples/wgpu-html5/example.js:3]: Use the simpler fetch API to read files.

File: examples/wgpu-html5/example.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            pass.setPipeline(pipeline);
+            const vertexBufferSlot = 0;
+            pass.setVertexBuffer(vertexBufferSlot, vertexBuffer);
+            pass.draw(vertices.length / 2);
+            pass.end();
+            const commandBuffer = encoder.finish();
+            device.queue.submit([commandBuffer]);
+        }
+
+        Promise.all([
```
Comment (csyonghe) [inline examples/wgpu-html5/example.js:105]: Use async/await instead of explicit promise.

File: examples/wgpu-html5/example.js (hunk 1 above)
Comment (aleino-nv) [reply examples/wgpu-html5/example.js:3]: Done. Thanks, I wasn't aware of the `fetch` API.

File: examples/wgpu-html5/example.js (hunk 2 above)
Comment (aleino-nv) [reply examples/wgpu-html5/example.js:105]: You mean like this?

    var vertexShaderResponse = await fetch("shader.vertex.wgsl");
    var vertexShaderCode = await vertexShaderResponse.text();
    var fragmentShaderResponse = await fetch("shader.fragment.wgsl");
    var fragmentShaderCode = await fragmentShaderResponse.text();

That would load the shaders serially one at a time, which is why I opted for `Promise.all` which I believe allows you to overlap the fetches.
Not a big deal for this app though, so I'll change it if you want.

Comment (aleino-nv) [reply examples/wgpu-html5/example.js]: Done. The last commit switches to serial `await`.

**Review by csyonghe — APPROVED** [review]

**Pull request merged**